### PR TITLE
Add corner test cases for filtering in InstructorStudentListPage #4280

### LINKED ISSUE #4280: Add corner test cases for filtering in InstructorStudentListPage
Add test cases for testing if the prefix id check for team and section is correct

### PR DESCRIPTION
**Pull request by imouto1994**
Fixed #4280

### CODE REVIEW

File: src/test/javascript/InstructorStudentListJsTest.js [hunk 1]
```diff
@@ -41,7 +41,7 @@ test('filterSection()', function(){
 
     equal($("[id^=student_email]:visible").length, 12, "All emails (minus duplicate) should be visible again");
 
-     // Deselect 1 section and select back usin Select All
+     // Deselect 1 section and select back by select that section
```
Comment (LowWeiLin) [inline src/test/javascript/InstructorStudentListJsTest.js]: ...by selecting that section

File: src/test/javascript/InstructorStudentListJsTest.js [hunk 2]
```diff
@@ -64,9 +64,10 @@ test('filterSection()', function(){
     equal($("#studentsection-c0\\.0").is(':visible'), true, "All sections should be visible");
     equal($("#studentsection-c0\\.1").is(':visible'), true, "All sections should be visible");
     equal($("#studentsection-c1\\.0").is(':visible'), true, "All sections should be visible");
-    equal($("#section_check-0-0").is(':checked'), true, "Course 2 Section A should be re-selected");
     equal($("#section_check-0-1").is(':checked'), true, "Course 2 Section B should be re-selected");
-    equal($("#section_check-1-0").is(':checked'), true, "Course 3 Section C should be re-selected");
+    equal($("#team_check-0-1-0").is(':checked'), true, "Course 2 Section B Team 2 should be selected");
+    equal($("#team_check-0-1-1").is(':checked'), true, "Course 2 Section B Team 3 should be selected");
+    equal($("#team_check-0-10-0").is(':checked'), true, "Course 2 Invalid Section Team should not be selected");
```
Comment (LowWeiLin) [inline src/test/javascript/InstructorStudentListJsTest.js]: Should this be true or false? The text says it should not be selected, but the assertion is otherwise?

File: src/test/javascript/InstructorStudentListJsTest.js [hunk 3]
```diff
@@ -64,9 +64,10 @@ test('filterSection()', function(){
     equal($("#studentsection-c0\\.0").is(':visible'), true, "All sections should be visible");
     equal($("#studentsection-c0\\.1").is(':visible'), true, "All sections should be visible");
     equal($("#studentsection-c1\\.0").is(':visible'), true, "All sections should be visible");
-    equal($("#section_check-0-0").is(':checked'), true, "Course 2 Section A should be re-selected");
     equal($("#section_check-0-1").is(':checked'), true, "Course 2 Section B should be re-selected");
-    equal($("#section_check-1-0").is(':checked'), true, "Course 3 Section C should be re-selected");
```
Comment (LowWeiLin) [inline src/test/javascript/InstructorStudentListJsTest.js]: Why are these lines removed?

Comment (imouto1994) [inline src/test/javascript/InstructorStudentListJsTest.js]: Deselect section B does not affect section C so no need to check right?

Comment (LowWeiLin) [inline src/test/javascript/InstructorStudentListJsTest.js]: Okay, would make sense if it was checking that it won't be affected, but its not doing that.

File: src/test/javascript/InstructorStudentListJsTest.js (hunk 2 above)
Comment (imouto1994) [inline src/test/javascript/InstructorStudentListJsTest.js]: The test passed as the element is checked due to select all. I have fixed it to be correct :)

File: src/test/javascript/InstructorStudentListJsTest.js (hunk 1 above)
Comment (imouto1994) [inline src/test/javascript/InstructorStudentListJsTest.js]: Updated :)

**Comment by LowWeiLin** [comment]
Okay can merge!

**Comment by LowWeiLin** [comment]
Make sure to run and ensure the test passes
